Foundation/NSURL.swift

```diff
@@ -957,7 +957,7 @@ extension NSURL {
     
     open func appendingPathComponent(_ pathComponent: String) -> URL? {
         var result : URL? = appendingPathComponent(pathComponent, isDirectory: false)
-        if !pathComponent.hasSuffix("/") && isFileURL {
+        if !validPathSeps.contains(where: { pathComponent.hasSuffix(String($0)) }) && isFileURL {
```

Comment:
So;

Foundation distinguishes between paths (in `NSString`s) and `URL`s. The latter have harmonized behavior between all platforms, whereas the former can contain a platform path and can thus accommodate specialized platform behaviors.

Trying to append a `\` to an URL is a programmer error regardless of platform. URL paths are specified to use `/` as separators. Thus: I would back off this line and I'm okay taking the rest.

Reply:
The only bit that puts a wrench into this particular getup is that the result of `URL`'s `.path` _must_ be useable as a platform path, which can get complicated.

Reply:
> The only bit that puts a wrench into this particular getup is that the result of `URL`'s `.path` _must_ be useable as a platform path, which can get complicated.

Looks like `/` path separator will meet this requirement for all platforms, as Windows is (more or less) loyal to both variants. But can you please add more on this "must" part? I always thought that URL/CFURL api clearly distinguish between _file system_ paths and URL paths. Now I feel like I need to reevaluate this for myself.

Reply:
> The only bit that puts a wrench into this particular getup is that the result of `URL`'s `.path` _must_ be useable as a platform path, which can get complicated.

 I see, that is complicated. What are your thoughts on handling absolute URL paths then? On Windows, paths which begin with `/` or `\` are drive relative. That is, if my cwd is `C:\` then `/Users/gwenm` resolves to `C:\Users\gwenm`.

Unfortunately, `/C:/Users/gwenm` is not a valid windows path, so I think there really only two reasonable things to return from `URL(fileURLWithPath: "C:\\Users\\gwenm").path` on Windows:
- `/C:/Users/gwenm` which, while not a valid Windows path, we would at least be able to add support for through out Foundation so that e.g. `"foo".write(to: myURL.path, ...)` would work.
- `C:/Users/gwenm` which isn't a proper RFC 1808 path as specified to be returned by Foundation, as an absolute path _must_ start with `/`,  but would work directly with Windows apis (though would not be able to be copied and pasted into cmd.exe and directly used without first being quoted).